connections/ws.go

```diff
@@ -249,14 +259,26 @@ func WSStreamProto[T proto.Message](w *WS, ctx context.Context, streamName strin
 }
 
 func wsStream[T any](w *WS, ctx context.Context, streamName string, streamParams json.RawMessage, unmarshal func(b []byte) (T, error)) (Streamer[T], error) {
-	params := SubscribeParams{
-		StreamName: streamName,
-		StreamOpts: streamParams,
-	}
-	paramsB, err := json.Marshal(params)
-	if err != nil {
-		return nil, err
+	var (
+		err     error
+		paramsB []byte
+	)
+	if streamParams == nil {
+		paramsB, err = json.Marshal([]string{streamName})
```

Comment:
slightly better would be to edit `SubscribeParams`'s `MarshalJSON` function to check if `streamOpts` is nil, and to not marshal it there.